File: exercism/csharp/nth-prime/NthPrime.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class NthPrime
{
    public static int Nth (int n)
    {
        var primes = new HashSet<int>();
        var p = 2;
        while (true) {
            var isPrime = true;
            foreach (var q in primes) {
                if (p % q == 0) {
                    p++;
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                primes.Add(p);
                if (primes.Count() == n) return p;
                p++;
            }
        }
    }
}

File: exercism/c/nth-prime/src/nth_prime.c
#include "nth_prime.h"

int nth(int n) {
  if (n == 0)
    return 0;

  int primes[n], k = 0, p = 2, is_prime;

  while (k < n) {
    is_prime = 1;
    for (int i = 0; i < k; i++) {
      if (p % primes[i] == 0) {
        p++;
        is_prime = 0;
        break;
      }
    }
    if (is_prime) {
      primes[k++] = p++;
    }
  }

  return primes[n - 1];
}
